### PR TITLE
[Form] Add `group_by` option for `EnumType`

### LINKED ISSUE #17707: [Form] `EnumType`: `group_by` is missing
`group_by` is missing on https://symfony.com/doc/current/reference/forms/types/enum.html

I'm not sure if the text from https://symfony.com/doc/current/reference/forms/types/choice.html#group-by can be reused, cause I don't know how this multi-dimensional array thing would work:

> You can group the <option> elements of a `<select>` into `<optgroup>` by passing a multi-dimensional array to choices

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Resolves https://github.com/symfony/symfony-docs/issues/17707

### CODE REVIEW

**Review by OskarStark — APPROVED** [review]

File: reference/forms/types/enum.rst [hunk 1]
```diff
@@ -92,6 +92,59 @@ These options inherit from the :doc:`ChoiceType </reference/forms/types/choice>`
...
+                return 'Lower';
+            }
+
+            return 'Other';
+        }
+    ]);
+
+This callback will group choices in 3 categories: ``Upper``, ``Lower`` and ``Other``.
+
+If you return ``null``, the option won't be grouped.
```
Comment (OskarStark) [inline reference/forms/types/enum.rst:146]: versionadded missing

Comment (alexandre-daubois) [reply reference/forms/types/enum.rst:146]: Is it worth it, as it was actually added at the same time as the EnumType itself?

Comment (OskarStark) [reply reference/forms/types/enum.rst:146]: Not needed then, thanks

**Comment by javiereguiluz** [comment]
Alex, thanks a lot for this contribution. I loved how you adapted the existing example; it feels very natural. Thanks!

PS: I merged in 5.4 and up because EnumType was introduced in 5.4.

**Pull request merged**